rust/gateway/src/main.rs

```diff
@@ -31,17 +33,10 @@ async fn main() -> Result<()> {
         SecretString::new(cli.common.token),
         cli.common.firezone_id,
     )?;
-    let tunnel = Arc::new(Tunnel::new(private_key, CallbackHandler).await?);
 
-    let task = pin!(backoff::future::retry_notify(
-        ExponentialBackoffBuilder::default()
-            .with_max_elapsed_time(None)
-            .build(),
-        move || run(tunnel.clone(), connect_url.clone()).map_err(to_backoff),
-        |error, t| {
-            tracing::warn!(retry_in = ?t, "Error connecting to portal: {error:#}");
-        },
-    ));
+    let task =
+        tokio::spawn(async move { run_loop(connect_url, private_key).await }).map_err(Into::into);
```

Comment:
There shouldn't be a need for `async move` if you don't capture any references in the parameters.